libsolidity/formal/SMTEncoder.cpp

```diff
@@ -883,6 +886,102 @@ void SMTEncoder::endVisit(ElementaryTypeNameExpression const& _typeName)
 	m_context.createExpression(_typeName, result.second);
 }
 
+namespace
+{
+unsigned dimensionCount(TypePointer _type)
+{
+	unsigned dimensionCount = 0;
+	while (_type->category() == Type::Category::Array || _type->category() == Type::Category::Mapping)
+	{
+		++dimensionCount;
+		if (_type->category() == Type::Category::Array)
+			_type = dynamic_cast<ArrayType const *>(_type)->baseType();
+		else if (_type->category() == Type::Category::Mapping)
+			_type = dynamic_cast<MappingType const *>(_type)->valueType();
+		else
+			solAssert(false, "");
+	}
+	return dimensionCount;
+}
+
+vector<string> structGetterReturnedMembers(StructType const* _structType)
+{
+	vector<string> returnedMembers;
+	for (auto const& member: _structType->nativeMembers(nullptr))
+	{
+		// TODO: Figure out which types are returned by the getter and which are ignored
+		if (auto category = member.type->category(); category == Type::Category::Array || category == Type::Category::Mapping)
```

Comment:
```suggestion
		if (
		    auto category = member.type->category();
		    category == Type::Category::Array || category == Type::Category::Mapping
		)
```

Reply:
`string` and `bytes` are fine, right? In that case we'd need an extra condition that if it's an array, it's not `string` nor `bytes`. Also need to check for static arrays

Reply:
True. I added special check for `string` and `byte`. Static arrays are handled the same way as dynamic arrays.